File: src/test/java/zhelonin/hm3/it/repo/TeacherRepositoryTest.java
package zhelonin.hm3.it.repo;

import static org.junit.jupiter.api.Assertions.*;

import java.util.List;
import java.util.Optional;
import lombok.RequiredArgsConstructor;
import org.junit.jupiter.api.Test;
import zhelonin.hm3.entity.Teacher;
import zhelonin.hm3.it.IntegrationBaseTest;
import zhelonin.hm3.repo.boot.TeacherRepository;


@RequiredArgsConstructor
class TeacherRepositoryTest extends IntegrationBaseTest {

    private final TeacherRepository teacherRepository;


    @Test
    void findById(){
      Optional<Teacher> byId = teacherRepository.findById(3);
      assertNotNull(byId);
    }

    @Test
    void delete(){
      teacherRepository.deleteById(2);
      assertTrue(teacherRepository.findById(2).isEmpty());
    }

    @Test
    void save(){
      Teacher teacher = new Teacher(null, "Ser");
      teacherRepository.save(teacher);
      assertNotNull(teacher.getId());
    }

    @Test
    void findAll(){
      List<Teacher> all = teacherRepository.findAll();
      assertEquals(3, all.size());
    }
}

File: src/test/java/zhelonin/hm3/it/repo/TeacherRepositoryTest.kt
package zhelonin.hm3.it.repo

import lombok.RequiredArgsConstructor
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import zhelonin.hm3.entity.Teacher
import zhelonin.hm3.it.IntegrationBaseTest
import zhelonin.hm3.repo.boot.TeacherRepository


internal class TeacherRepositoryTest : IntegrationBaseTest() {
    @Autowired
    private lateinit var teacherRepository: TeacherRepository


    @Test
    fun findById() {
        val byId = teacherRepository!!.findById(3)
        Assertions.assertNotNull(byId)
    }

    @Test
    fun delete() {
        teacherRepository!!.deleteById(2)
        Assertions.assertTrue(teacherRepository.findById(2).isEmpty)
    }

    @Test
    fun save() {
        val teacher = Teacher(null, "Ser")
        teacherRepository!!.save(teacher)
        Assertions.assertNotNull(teacher.id)
    }

    @Test
    fun findAll() {
        val all = teacherRepository!!.findAll()
        Assertions.assertEquals(3, all.size)
    }
}
